### PR TITLE
Stacked bar chart in plots tab for discrete v discrete data

### PR DESCRIPTION
**Pull request by adamabeshouse**
The colors are chosen by following process:
- If it is a value that we have a color for, e.g. one of the study view clin attr colors, or oncoprint mutation/cna colors, then we use it,
- Otherwise, we select from a preset list of colors (same process as in oncoprint)

image

image

image

image

image

image

image

### CODE REVIEW

File: src/shared/components/plots/PlotUtils.ts [hunk 1]
```diff
@@ -107,6 +107,57 @@ export function scatterPlotSize(
     }
 }
 
+function darkenHexChannel(c:string) {
```
Comment (alisman) [inline src/shared/components/plots/PlotUtils.ts]: should we be using a library for this?   open for discussion

File: src/shared/components/plots/StackedBarPlot.tsx [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
+            return 23.7*numRows;
+        }
+    }
+
+    @computed get legendData() {
+        return this.data.map(obj=>({
+            name: obj.minorCategory,
+            symbol:{
+                type: "square",
+                fill: this.getColor(obj.minorCategory),
```
Comment (alisman) [inline src/shared/components/plots/StackedBarPlot.tsx]: this seems slightly weird.  see above

File: src/shared/components/plots/PlotUtils.ts (hunk 1 above)
Comment (adamabeshouse) [reply src/shared/components/plots/PlotUtils.ts]: for which part?

File: src/shared/components/plots/StackedBarPlot.tsx [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+        this.container = container;
+    }
+
+    @computed get getColor() {
+        const uniqueColorGetter = makeUniqueColorGetter(_.values(this.props.categoryToColor));
+        const categoryToColor:{[category:string]:string} = {};
+        _.forEach(this.props.categoryToColor, (color, category)=>{
+            categoryToColor[category.toLowerCase()] = color;
+        });
+        return function(category:string) {
```
Comment (alisman) [inline src/shared/components/plots/StackedBarPlot.tsx]: i think you could make this function a method of the class the references a computed value categoryToColor.   it's maybe a little clearer?

Comment (adamabeshouse) [reply src/shared/components/plots/StackedBarPlot.tsx]: i dont understand your sentence, maybe some typos?

File: src/shared/components/plots/PlotUtils.ts (hunk 1 above)
Comment (alisman) [reply src/shared/components/plots/PlotUtils.ts]: all of this color manipulation stuff

**Comment by adamabeshouse** [comment]
@alisman approve?

**Pull request merged**